tasks/collector.py

```diff
@@ -36,6 +59,86 @@
 }
 
 
+class YAMLValidationError(Exception):
+    def __init__(self, message):
+        super().__init__(message)
+
+
+def find_matching_components(manifest, components_to_match: dict, present: bool) -> list:
+    """Given a manifest and dict of components to match, if present=True, return list of
+    components found, otherwise return list of components missing."""
+    res = []
+    for component_type, components in components_to_match.items():
+        for component in components:
+            found_component = False
+            components_matching_component_type = manifest.get(component_type)
+            if components_matching_component_type:
+                for module in components_matching_component_type:
+                    if module.get("gomod").find(component) != -1:
+                        found_component = True
+                        if not present:
+                            found_component = True
+                        else:
+                            res.append(component)
+                        break
+            if not present and not found_component:
+                res.append(component)
+    return res
+
+
+def validate_manifest(manifest) -> list:  # return list of components to remove, empty list if valid
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏Prefer using docstrings (`""" ... """`)

Reply:
have made the suggested change. thank you.